src/main/java/io/phasetwo/keycloak/email/OverridableEmailSenderProvider.java

```diff
@@ -0,0 +1,104 @@
+package io.phasetwo.keycloak.email;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import lombok.extern.jbosslog.JBossLog;
+import org.infinispan.Cache;
+import org.keycloak.connections.infinispan.InfinispanConnectionProvider;
+import org.keycloak.email.DefaultEmailSenderProvider;
+import org.keycloak.email.EmailException;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.UserModel;
+
+@JBossLog
+public class OverridableEmailSenderProvider extends DefaultEmailSenderProvider {
+
+  private final KeycloakSession session;
+  private final Map<String, String> conf;
+  private final Integer maxEmails;
+  private final String cacheKey;
+  private Cache<String, Integer> counterCache;
+
+  public OverridableEmailSenderProvider(
+      KeycloakSession session, Map<String, String> conf, Integer maxEmails) {
+    super(session);
+    this.session = session;
+    this.conf = conf;
+    this.maxEmails = maxEmails;
+    this.cacheKey = getCacheKey();
+    try {
+      this.counterCache =
+          session.getProvider(InfinispanConnectionProvider.class).getCache("counterCache", true);
+    } catch (Exception e) {
+      log.warnf("Error loading counterCache %s", e);
+    }
+  }
+
+  private boolean useOverride(Map<String, String> config) {
+    return (!config.isEmpty() && config.containsKey("host"));
+  }
+
+  private String getCacheKey() {
+    if (session.getContext().getRealm() != null) {
+      SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd");
+      return String.format(
+          "ext-email-override-emailCounter-%s-%s",
+          session.getContext().getRealm().getName(), formatter.format(new Date()));
+    } else {
+      return null;
+    }
+  }
+
+  private boolean canSend() {
+    if (cacheKey == null) return true;
+    Integer count = counterCache.get(cacheKey);
+    log.infof("Count for %s is %d / %d", cacheKey, count, maxEmails);
+    if (count == null || count <= maxEmails) return true;
+    else return false;
+  }
+
+  private Integer increment() {
+    if (cacheKey == null) return 0;
+    else
+      return counterCache.compute(
+          cacheKey, (key, value) -> (value == null) ? 1 : value + 1, 1, TimeUnit.DAYS);
+  }
+
+  @Override
+  public void send(
+      Map<String, String> config, UserModel user, String subject, String textBody, String htmlBody)
+      throws EmailException {
+    if (useOverride(config)) {
```

Comment:
It checks the `config` (= realm config) to see if it is populated. If it is, the realm config will "override" the global config. I think it's my language that's confusing, rather than needing to be negated.
"override" = realm config
"default" = global config

Reply:
I updated it to `useRealmConfig` for clarity.